cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/field/DefaultFieldLineageReader.java

```diff
@@ -112,7 +112,7 @@ public DefaultFieldLineageReader(DatasetFramework datasetFramework, TransactionS
         : fieldLineageDataset.getOutgoingOperations(endPointField.getEndPoint(), start, end);
     });
 
-    Set<ProgramRunOperations> endPointFieldOperations = new HashSet<>();
+    Set<ProgramRunOperations> endPointFieldOperations = new LinkedHashSet<>();
```

Comment:
why is this needed

Reply:
Added this to maintain the ordering within the different set of operations. This allows the returned set of operations to be sorted by time descending.

Reply:
is this just for convenience, or is this part of the contract. For example, does the UI rely on this behavior? If it is part of the contract, it should be documented in the javadocs.

Reply:
UI does not rely on this, since it displays whatever operations are returned by backend which is undefined. However its better to return them in the order. Added javadocs.